modules/redhen_connection/src/Plugin/ConnectionPermission/ConnectionConnectionPermission.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\redhen_connection\Plugin\ConnectionPermission;
+
+use Drupal\Core\Access\AccessResultNeutral;
+use Drupal\redhen_connection\Entity\ConnectionType;
+use Drupal\redhen_contact\Entity\Contact;
+use Drupal\redhen_connection\Plugin\ConnectionPermissionBase;
+use Drupal\redhen_connection\Plugin\ConnectionPermissionInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Provide permissions for the connection entity of a redhen_connection.
+ *
+ * @ConnectionPermission(
+ *  id = "connection_connection_permission",
+ *  label = @Translation("Connection"),
+ *  description = @Translation("Applies to both current users connection and secondary connections. Sitewide permissions will override this permission."),
+ *  subject_entity_type = "redhen_connection",
+ *  subject_entity_bundle = "",
+ *  influencer_entity_type = "",
+ * )
+ */
+class ConnectionConnectionPermission extends ConnectionPermissionBase implements ConnectionPermissionInterface {
+
+  /**
+   * A Redhen Contact Object.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  private $contact;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPermissionKey() {
+    return 'connection';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfluencers(EntityInterface $subject_entity) {
+    // Return the connections for the current contact and any.
+    return $this->redhenConnectionConnections->getConnectedEntities($this->contact, $subject_entity->getType());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function hasRolePermissions(EntityInterface $subject_entity, $operation, Contact $contact) {
+    $this->contact = $contact;
+    // Only check permissions for connections that are of a type with contacts.
+    $connection_type = ConnectionType::load($subject_entity->getType());
+    if ($connection_type->getEndpointEntityTypeId(1) == 'redhen_contact' || $connection_type->getEndpointEntityTypeId(2) == 'redhen_contact') {
+      $influencers = $this->getInfluencers($subject_entity);
+      if ($influencers) {
+        $influencer = reset($influencers);
```

Comment:
do we have the possibility of getting multiple responses here, or is this a case of just a nested array coming back with a single item every time? If we can have multiples, throwing the rest away seems like a problem.

Reply:
Good call, the case where I can see more than one influencer being returned here is if a contact has more than one of the same type of connection. In that scenario we should honor the role permissions from each connection. 

I updated the logic to iterate over the influencers and to return an isAllowed permission if found, otherwise it returns accessNeutral.

Reply:
Ah nice! Good thinking. Nice that we are doing affirmative access here, makes the loop easier to grock.